### PR TITLE
feat: add support for specifying what builder to use

### PR DESCRIPTION
**Pull request by kenneth-rosario**
The new `-builder-url` flag will help improve test coverage for oss builders that claim to support functions framework.

### CODE REVIEW

File: .github/workflows/buildpack-integration-test.yml [hunk 1]
```diff
@@ -27,6 +27,11 @@ on:
         type: string
         default: ''
         required: false
+      builder-url:
```
Comment (garethgeorge) [inline .github/workflows/buildpack-integration-test.yml:30]: nit: if the list of builders we support is consistent for each repository (I think it should be in most cases) as an alternative to exposing the `builder-url` such that each client repository needs to specify a list of urls, what if we make this a part of the configuration matrix below on line 131?

i.e. would
```
        type: [http, cloudevent, event]
        builder-url: ['', 'gcr.io/buildpacks/builder']
        include:
          - type: http
            builder-source: ${{ inputs.http-builder-source }}
            builder-target: ${{ inputs.http-builder-target }}
          - type: cloudevent
            builder-source: ${{ inputs.cloudevent-builder-source }}
            builder-target: ${{ inputs.cloudevent-builder-target }}
          - type: event
            builder-source: ${{ inputs.event-builder-source }}
            builder-target: ${{ inputs.event-builder-target }}
```
work?

**Review by garethgeorge — APPROVED** [review]

File: .github/workflows/buildpack-integration-test.yml (hunk 1 above)
Comment (kenneth-rosario) [reply .github/workflows/buildpack-integration-test.yml:30]: I like this idea. Will change this back to draft to implement and test this. Thanks!

**Comment by kenneth-rosario** [comment]
@garethgeorge I was able to add dynamic matrix capability. Unfortunately, I was not able to find GitHub Action support for plain string array on inputs.

**Comment by garethgeorge** [comment]
Sorry for the confusion @kenneth-rosario, hope I'm not making you go back and forth. Will the value need to be overridden? If it does, I think I prefer your original solution (I think the operation can be run in a matrix in whatever workflow invokes it?). Otherwise, I'm wondering if we can just hard code it in a central place (i.e. here in the conformance test repo) and do away with the input all together. 

Where possible I like the idea of centralizing our testing configuration such that the conformance tests running on all repos are as similar as possible.

**Comment by kenneth-rosario** [comment]
Ah I see, apologies for misunderstanding will go back to last implementation. I am not sure if its possible to have them all centralized since some of the builders are runtime specific. (e.g. "gcr.io/gae-runtimes/buildpacks/go/builder:latest", "gcr.io/gae-runtimes/buildpacks/google-gae-22/go/builder:latest") We might end up spawning too many runners that do not run (Or might fail).

**Review by garethgeorge — APPROVED** [review]

**Comment by garethgeorge** [comment]
Nice, thanks Kenneth!

**Pull request merged**